twine/repository.py

```diff
@@ -153,6 +153,11 @@ def upload(self, package, max_redirects=5):
         return resp
 
     def package_is_uploaded(self, package, bypass_cache=False):
+        # NOTE(sigmavirus24): Not all indices are PyPI and pypi.io doesn't
+        # have a similar interface for finding the package versions.
+        if not self.url.startswith('https://pypi.python.org/pypi/'):
```

Comment:
``if self.url.startswith(("https://pypi.python.org/", "https://upload.pypi.io/")):`` is what I would do here.